cl/corpus_importer/management/commands/harvard_merge.py

```diff
@@ -0,0 +1,895 @@
+
+        if (
+            cl_clean_docket in h_clean_docket
+            and cl_docket.docket_number != h_clean_docket
+        ):
+            cl_docket.docket_number = h_clean_docket
+            cl_docket.save()
+        else:
+            # Check if their relatively similar and if so save the harvard one
+            # if its longer
+            similarity = get_cosine_similarity(cl_clean_docket, h_clean_docket)
+            if similarity > 0.8:
+                if len(h_clean_docket) > len(cl_clean_docket):
+                    cl_docket.docket_number = h_clean_docket
+                    cl_docket.save()
+
+
+def merge_case_names(cluster_id: int, harvard_data: Dict[str, Any]) -> None:
+    """Merge case names
+
+    :param cluster_id: The cluster id of the merging item
+    :param harvard_data: json data from harvard case
+    :return: None
+    """
+    cluster = OpinionCluster.objects.get(id=cluster_id)
+    harvard_case_name = titlecase(harmonize(harvard_data["name_abbreviation"]))
+    harvard_case_name_full = titlecase(harvard_data["name"])
+    cluster_case_name = titlecase(harmonize(cluster.case_name))
+    cluster_case_name_full = titlecase(cluster.case_name_full)
+
+    update_dict = {}
+    # Case with full case names
+    if not cluster_case_name_full and harvard_case_name_full:
+        update_dict["case_name_full"] = harvard_case_name_full
+        # Change stored value to new
+        cluster_case_name_full = harvard_case_name_full
+    elif cluster_case_name_full and harvard_case_name_full:
+        if len(harvard_case_name_full) > len(cluster_case_name_full):
+            # Select best case name based on string length
+            update_dict["case_name_full"] = harvard_case_name_full
+            # Change stored value to new
+            cluster_case_name_full = harvard_case_name_full
+    else:
+        # We don't care if harvard data is empty or both are empty
+        pass
+
+    # Case with abbreviated case names
+    if not cluster_case_name and harvard_case_name:
+        update_dict["case_name"] = harvard_case_name
+        # Change stored value to new
+        cluster_case_name = harvard_case_name
+    elif cluster_case_name and harvard_case_name:
+        if len(harvard_case_name) > len(cluster_case_name):
+            # Select best case name based on string length
+            update_dict["case_name"] = harvard_case_name
+            # Change stored value to new
+            cluster_case_name = harvard_case_name
+    else:
+        # We don't care if harvard data is empty or both are empty
+        pass
+
+    if cluster_case_name_full and cluster_case_name:
+        if len(cluster_case_name) > len(cluster_case_name_full):
+            # Swap field values
+            update_dict["case_name"] = cluster_case_name_full
+            update_dict["case_name_full"] = cluster_case_name
+
+    if update_dict:
+        OpinionCluster.objects.filter(id=cluster_id).update(**update_dict)
```

Comment:
Again, we should see if we can do as many of our updates at the end as possible, rather than at the end of each function.

Reply:
I made a few changes decrease the number of updates for the opinion cluster